addons/edi/models/edi_gateway.py

```diff
@@ -105,6 +105,11 @@ class EdiGateway(models.Model):
     can_initiate = fields.Boolean(string="Initiate Connections",
                                   compute='_compute_can_initiate')
     server = fields.Char(string="Server Address")
+    outgoing_server = fields.Char(string="Outgoing Server Address")
+    # Map the server address to the desired server
+    outgoing_server_id = fields.Integer(
```

Comment:
* Since this depends on a search of specific model, we could use a Many2One field.
* Make it readonly